### PR TITLE
feat(dynamodb): Changes how metricForOperation methods are used

### LINKED ISSUE #21963: (aws-dynamodb): metricThrottledRequestsForOperation is not part of the ITable interface.
### Describe the bug

[ITable](https://github.com/aws/aws-cdk/blob/ab7668197b1d7d7d08654d49df0c5ad9cb61b960/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L367) interface doesn't contain [metricThrottledRequestsForOperation](https://github.com/aws/aws-cdk/blob/ab7668197b1d7d7d08654d49df0c5ad9cb61b960/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L877) but instead contains deprecated [metricThrottledRequests](https://github.com/aws/aws-cdk/blob/ab7668197b1d7d7d08654d49df0c5ad9cb61b960/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L538) which has an issue and doesn't provide any metrics at all. By just having metricThrottledRequests in ITable interface is hiding the fact that it's deprecated and not allowing the recommended metricThrottledRequestsForOperation to use for users who declare the objects with ITable as type.

### Expected Behavior

ITable interface should contain metricThrottledRequestsForOperation declaration

### Current Behavior

ITable interface contains [metricThrottledRequests](https://github.com/aws/aws-cdk/blob/ab7668197b1d7d7d08654d49df0c5ad9cb61b960/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L538) declartion which is providing invalid metrics or not providing any metircs at all. So there is a new method (metricThrottledRequestsForOperation) provided as alternative and recommended according to documentation but it's not part of the ITable interface. If an object is declared with ITable type then the recommended method can't be accesed.

```

  /**
   * How many requests are throttled on this table
   *
   * Default: sum over 5 minutes
   *
   * @deprecated Do not use this function. It returns an invalid metric. Use `metricThrottledRequestsForOperation` instead.
   */
  public metricThrottledRequests(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
    return this.metric('ThrottledRequests', { statistic: 'sum', ...props });
  }

  /**
   * How many requests are throttled on this table, for the given operation
   *
   * Default: sum over 5 minutes
   */
  public metricThrottledRequestsForOperation(operation: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
    return new cloudwatch.Metric({
      ...DynamoDBMetrics.throttledRequestsSum({ Operation: operation, TableName: this.tableName }),
      ...props,
    }).attachTo(this);
  }
```

### Reproduction Steps

if an object is declared with `ITable` as type then only `getThrottledMetrics` can be accessed and not `metricThrottledRequestsForOperation`

```
export class DynamoDBTableMetricGenerator {

    private readonly dynamoDbTable: ITable;

    constructor(dynamoDbTable: ITable) {
        this.dynamoDbTable = props.dynamoDbTable;
    }

    /**
     * Returns metircs for throttled requests on the table for all the operations passed
     */
    getThrottledMetrics(operations: strings[], props: MetricOptions): Metric[] {
         operations.forEach((operation) => {
                this.dynamoDbTable.metricThrottledRequests({
                    label: operation,
                    period: Duration.minutes(1),
                    dimensions: {
                        TableName: this.dynamoDbTable.tableName,
                        Operation: operation,
                    },
                }),
            });
        }
    }
```

### Possible Solution

If `metricThrottledRequestsForOperation` is added to the `ITable` interface then the object can access it and use accordingly

```
export class DynamoDBTableMetricGenerator {

    private readonly dynamoDbTable: ITable;

    constructor(dynamoDbTable: ITable) {
        this.dynamoDbTable = props.dynamoDbTable;
    }

    /**
     * Returns metircs for throttled requests on the table for all the operations passed
     */
    getThrottledMetrics(operations: strings[], props: MetricOptions): Metric[] {
         operations.forEach((operation) => {
                this.dynamoDbTable.metricThrottledRequestsForOperation(operation, {
                    label: operation,
                    period: Duration.minutes(1),
                    dimensions: {
                        TableName: this.dynamoDbTable.tableName,
                        Operation: operation,
                    },
                }),
            });
        }
    }
```

### Additional Information/Context

_No response_

### CDK CLI Version

3

### Framework Version

_No response_

### Node.js Version

14.x

### OS

GNU/Linux

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by marciocadev**
# Description

This PR is changing `metricThrottledRequestsForOperation` and `metricSystemErrorForOperations` to receive the `OperationsMetricsOptions` parameter to standardize these calls and adds `metricThrottledRequestsForOperation` to the ITable interface

I also change the `metricThrottledRequests` to deprecated

Closes #21963 
----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-dynamodb/lib/table.ts [hunk 1]
```diff
@@ -904,28 +901,49 @@ abstract class TableBase extends Resource implements ITable {
...
   }
 
+  /**
+   * How many requests are throttled on this table.
+   *
+   * This will sum errors across all possible operations.
+   * Note that by default, each individual metric will be calculated as a sum over a period of 5 minutes.
+   * You can customize this by using the `statistic` and `period` properties.
+   */
+  public metricThrottledRequestsForOperation(props?: OperationsMetricOptions): cloudwatch.IMetric {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-dynamodb/lib/table.ts:911]: We can't change the contract for this method. We could add a new one
`metricThrottledRequestsForOperations` (plural).

File: packages/@aws-cdk/aws-dynamodb/lib/table.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ const MAX_LOCAL_SECONDARY_INDEX_COUNT = 5;
 /**
  * Options for configuring a system errors metric that considers multiple operations.
  */
-export interface SystemErrorsForOperationsMetricOptions extends cloudwatch.MetricOptions {
+export interface OperationsMetricOptions extends cloudwatch.MetricOptions {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-dynamodb/lib/table.ts:28]: I don't think we can rename this interface. I think I might prefer having a second input prop for
operations.

```ts
metricThrottledRequestsForOperations(operations: string[], props?: cloudwatch.MetricOptions):
cloudwatch.IMetric;
```

**Review by corymhall — CHANGES_REQUESTED** [review]
Just a couple of comments

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

File: packages/@aws-cdk/aws-dynamodb/README.md [hunk 3]
```diff
@@ -208,3 +208,26 @@ const table = new dynamodb.Table(this, 'Table', {
   kinesisStream: stream,
 });
 ```
+
+## Alarm metrics
+
+Alarms can be configurated on the DynamoDB table to captured metric data
```
[the remark this answers is not in the document]
Comment (marciocadev) [reply packages/@aws-cdk/aws-dynamodb/README.md:214]: thank you

File: packages/@aws-cdk/aws-dynamodb/lib/table.ts (hunk 2 above)
Comment (marciocadev) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts:28]: I can reverce the interface but I think creating another interface for throttle will be better to keep a pattern

Comment (corymhall) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: I do like the API with the `operations` as a separate input property better, but I also agree with you that it is better to keep the current patter.

**Review by corymhall — APPROVED** [review]
Thanks for the PR!

**Pull request merged**